Make anne-bonny and bonnyci[bot] equivalent

jamielennox opened the pull request:
In the BonnyCI migration to using integrations the role of anne-bonny
will hereafter be played by bonnyci[bot]. To ease this transition treat
any reference in config or status to anne-bonny as equivalent to
bonnyci[bot].

This is not a long term good idea, but will let us do the transition and
revert this patch in a week or so.

Change-Id: I9b144c2b978eb82fcee42c08c60cde8142abd02a
Signed-off-by: Jamie Lennox <jamielennox@gmail.com>

jamielennox committed:
Make anne-bonny and bonnyci[bot] equivalent

In the BonnyCI migration to using integrations the role of anne-bonny
will hereafter be played by bonnyci[bot]. To ease this transition treat
any reference in config or status to anne-bonny as equivalent to
bonnyci[bot].

This is not a long term good idea, but will let us do the transition and
revert this patch in a week or so.

Change-Id: I9b144c2b978eb82fcee42c08c60cde8142abd02a
Signed-off-by: Jamie Lennox <jamielennox@gmail.com>

anne-bonny commented:
Build failed.

- http://logs.bonnyci.org/check_github/BonnyCI/zuul/39/1491863276.28/bonnyci-run-check : FAILURE in 3m 10s

omgjlk commented on zuul/trigger/github.py:
```diff
@@ -32,6 +32,9 @@ def toList(item):
 
         efilters = []
         for trigger in toList(trigger_config):
+            statuses = [s.replace('anne-bonny', 'bonnyci[bot]'
```
Missing a closing paren on the `replace()` call.

omgjlk reviewed (commented):
I like the intent here, and I agree that it'll be the smoothest transition. It'll require two restarts, but those are easier to handle than coordinating a flag day of repo configuration changes.

anne-bonny commented:
Build failed.

- http://logs.bonnyci.org/check_github/BonnyCI/zuul/39/1491864354.85/bonnyci-run-check : FAILURE in 3m 10s

jamielennox committed:
If installation_key is not found revert to API key

Hacks on hacks. We already maintain a map of projects that are using an
integration so that we can fetch the installation id. If the project is
not in this list it's probably because they are not using an integration
and so we should fall back to using an api key.

Change-Id: I1fcfb75c93de0a34e10eead6f713087ea41d3586
Signed-off-by: Jamie Lennox <jamielennox@gmail.com>

anne-bonny commented:
Build failed.

- http://logs.bonnyci.org/check_github/BonnyCI/zuul/39/1491864777.01/bonnyci-run-check : FAILURE in 3m 08s

anne-bonny commented:
Build failed.

- http://logs.bonnyci.org/check_github/BonnyCI/zuul/39/1492471460.42/bonnyci-run-check : FAILURE in 3m 10s

anne-bonny commented:
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/39/1492473106.38/bonnyci-run-gate : FAILURE in 35m 46s

jamielennox commented:
reverify

jamielennox commented:
gate-recheck
